### PR TITLE
[SPARK-49874][SQL] Remove trim and ltrim collation specifiers.

### PR DESCRIPTION
**Pull request by jovanpavl-db**
### What changes were proposed in this pull request?
Removing collation trim and rtrim support.

### Why are the changes needed?
Trim and Ltrim collations won't be initially supported.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Changed existing tests.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java [hunk 1]
```diff
@@ -275,7 +275,7 @@ protected enum ImplementationProvider {
        * remained reserved (and fixed to 0) for future use.
        */
       protected enum SpaceTrimming {
-        NONE, LTRIM, RTRIM, TRIM
```
Comment (uros-db) [inline common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java:278]: update the comment above SpaceTrimming

File: common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java [hunk 2]
```diff
@@ -217,7 +217,7 @@ public Collation(
      * bit 31-24: Zeroes.
      * bit 23-22: Zeroes, reserved for version.
      * bit 21-18: Reserved for space trimming.
-     * 0000 = none, 0001 = left trim, 0010 = right trim, 0011 = trim.
+     * 0000 = none, 0001 = right trim. Bits 21-19 remain reserved and fixed to 0.
```
Comment (mihailom-db) [inline common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java:220]: nit: Could you separate this in two parts. bit 18 and then 0 and 1 and 21-19 reserved. For readability.

**Review by mihailomilosevic2001 — APPROVED** [review]
lgtm, some nit changes

**Review by uros-db — APPROVED** [review]
some nits, otherwise lgtm

File: common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java (hunk 2 above)
Comment (jovanpavl-db) [reply common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java]: Done.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @jovanpavl-db and @uros-db @mihailom-db for review.

**Pull request closed without merging**